src/configobj/__init__.py

```diff
@@ -472,6 +472,8 @@ def _initialise(self):
         self.default_values = {}
         self.extra_values = []
         self._created = False
+        # for lines
+        self.lines = {}
```

Comment:
a better name would be `source_map` and the comment is redundant.

generally, your PR is missing docs + tests.